CHANGELOG.md

```diff
@@ -1,30 +1,48 @@
 # Changelog
+All notable changes to this project will be documented in this file.
 
-## [Version 0.3.0 (Public Preview)](https://github.com/awslabs/aws-mysql-jdbc/releases/tag/0.3.0) - 2021-11-18
+The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/), and this project adheres to [Semantic Versioning](https://semver.org/#semantic-versioning-200).
+
+## [0.4.0] - 2021-12-14
+
+### Added
+* Connection plugin support through Connection Plugin Manager. More details can be found [here](https://github.com/awslabs/aws-mysql-jdbc#connection-plugin-manager).
+* [Enhanced Failure Monitoring](https://github.com/awslabs/aws-mysql-jdbc/#enhanced-failure-monitoring) for more sophisticated connection failure detection.
+* Tutorial on writing custom connection plugin can be found [here](https://github.com/awslabs/aws-mysql-jdbc/tree/main/src/demo/java/customplugins).
```

Comment:
I think having the tutorial here is a bit awkward given that it's not a code change in the actual driver. Could we link this to the first point? Tutorial on creating custom connection plugins can be found [here](https://github.com/awslabs/aws-mysql-jdbc/tree/main/src/demo/java/customplugins).